xbmc/network/httprequesthandler/HTTPPythonHandler.cpp

```diff
@@ -0,0 +1,242 @@
+/*
+ *      Copyright (C) 2015 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "HTTPPythonHandler.h"
+#include "URL.h"
+#include "interfaces/generic/ScriptInvocationManager.h"
+#include "interfaces/python/XBPython.h"
+#include "filesystem/File.h"
+#include "network/WebServer.h"
+#include "network/httprequesthandler/HTTPWebinterfaceHandler.h"
+#include "network/httprequesthandler/python/HTTPModPythonInvoker.h"
+#include "network/httprequesthandler/python/HTTPPythonInvoker.h"
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+
+#define MAX_STRING_POST_SIZE 20000
+
+CHTTPPythonHandler::CHTTPPythonHandler()
+  : IHTTPRequestHandler(),
+    m_scriptPath(),
+    m_addon(),
+    m_type(),
+    m_lastModified(),
+    m_requestData(),
+    m_responseData(),
+    m_responseRanges()
+{ }
+
+CHTTPPythonHandler::CHTTPPythonHandler(const HTTPRequest &request)
+  : IHTTPRequestHandler(request),
+    m_scriptPath(),
+    m_addon(),
+    m_type(),
+    m_lastModified(),
+    m_requestData(),
+    m_responseData(),
+    m_responseRanges()
+{
+  // get the real path of the script and check if it actually exists
+  m_response.status = CHTTPWebinterfaceHandler::ResolveUrl(m_request.pathUrl, m_scriptPath, m_addon);
```

Comment:
And here. In WSGI this path should dispatch to the same application. I.e `m_scriptPath` should always be `m_addon.LibPath()` for instance.